src/app/app.component.ts

```diff
@@ -21,4 +26,37 @@ export class AppComponent {
       return 'off-home';
     }
   }
+
+  ngDoCheck() {
+    if (this.auth.authenticated() && this.currentUserId == null)
```

Comment:
could this be accomplished without `DoCheck`? maybe another lifecycle event, preferably one called only once like `onInit`?